Neuro/Patches/PlayerControl_Patches.cs

```diff
@@ -19,4 +20,27 @@ public static void Postfix(PlayerControl __instance)
     {
         if (PlayerControl.LocalPlayer == __instance) PluginSingleton<NeuroPlugin>.Instance.FixedUpdate(__instance);
     }
+}
+
+[HarmonyPatch(typeof(PlayerControl), nameof(PlayerControl.Die))]
+public static class PlayerControl_Die
+{
+    public static void Postfix(DeathReason reason, PlayerControl __instance)
+    {
+        // Check if player was murdered
+        if (reason == DeathReason.Kill)
+        {
+            // Find dead body at death position
+            Collider2D[] deathPointColliders = Physics2D.OverlapCircleAll(__instance.GetTruePosition(), 1f);
+            foreach (Collider2D col in deathPointColliders)
+            {
+                // Call vision method for handling new dead bodies
+                DeadBody deadBody = col.gameObject.GetComponent<DeadBody>();
+                if (deadBody)
+                {
+                    PluginSingleton<NeuroPlugin>.Instance.vision.DeadBodyAppeared(deadBody);
+                }
```

Comment:
The reason I replaced FindObjectsOfType is because it is noted in documentation that this function is very slow. It could be used here now since I moved this logic out of FixedUpdate and it won't be run every frame anymore, but won't it still affect the performance though?

Reply:
No, it won't affect performance at all now that it's out of the update loop